### PR TITLE
feat: add logger components

### PR DESCRIPTION
**Pull request by stropitek**
Closes: https://github.com/zakodium-oss/react-science/issues/449

### CODE REVIEW

**Comment by stropitek** [comment]
Usage:

```jsx
import { FifoLogger } from 'fifo-logger';
const fifoLogger = new FifoLogger();

function App() {
  return (
    <FifoLoggerProvider logger={fifoLogger}>
      // ...
      <FifoLoggerToolbarItem />
    </FifoLoggerProvider>
  );
}
```

To add logs:

```jsx
function ComponentAddingLogs() {
  const logger = useFifoLogger();

  // In callback / useEffect
  logger.info('info');
}
```

To get render something when the logs are updated:

```jsx
function ComponentThatNeedsLogs() {
  const logs = useFifoLogs();
}
```

**Comment by stropitek** [comment]
@lpatiny This PR is awaiting your review

@moonayyur can you have a look as well? Because you designed a similar component in pixelium and the goal is to replace it with this one.

**Review by lpatiny — APPROVED** [review]
Functionality LGTM but there is no storybook so it may be difficult to find this component,

**Review by moonayyur — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -60,6 +60,7 @@
   "peerDependencies": {
     "@blueprintjs/core": "^5.3.2",
     "@blueprintjs/icons": "^5.1.7",
+    "fifo-logger": "^1.0.0",
```
Comment (targos) [inline package.json:63]: I'm not sure why it should be a peerDependency?

Comment (targos) [reply package.json:63]: Because only the types are used.

**Pull request merged**